engine/table/src/main/java/io/deephaven/engine/table/impl/BaseTable.java

```diff
@@ -630,6 +630,9 @@ public final void notifyListeners(final TableUpdate update) {
             Assert.assertion(getRowSet().sizePrev() == 0 || getRowSet().lastRowKeyPrev() < update.added().firstRowKey(),
                     "getRowSet().lastRowKeyPrev() < update.added().firstRowKey()");
         }
+        if (isStream()) {
+            Assert.assertion(update.shifted().empty(), "update.shifted.empty()");
```

Comment:
We could also assert:
1. No modifies
2. Removed size == previous size (other validation and rules ensure removed is a subset of previous)
3. Added size == current size (other validation and rules ensure added is a subset of current)
```suggestion
            Assert.eq(update.added().size(), "added size", getRowSet().size(), "current table size");
            Assert.eq(update.removed().size(), "removed size", getRowSet().sizePrev(), "previous table size");
            Assert.assertion(update.modified().isEmpty(), "update.modified.isEmpty()");
            Assert.assertion(update.shifted().empty(), "update.shifted.empty()");
```